### PR TITLE
178811031 old commission changes

### PR DESCRIPTION
**Pull request by maciejkocylapc**
#### What's this PR do?
script to add historical commission changes

#### How should this be manually tested?
rails r script/back_fill_commission_history.rb

#### What are the relevant tickets?
https://www.pivotaltracker.com/story/show/178811031

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

**Comment by kbiala** [comment]
@maciejkocylapc We'll need to gather network as well. Please merge last updates from master (migration with network is already there) and update your script.

File: app/controllers/public_controller.rb [hunk 1]
```diff
@@ -103,6 +103,18 @@ def contact_us
     @title = t('public.contact_us.title')
   end
 
+  def commission_histories
+    if params[:validator_id]
```
Comment (radek-polcode) [inline app/controllers/public_controller.rb:107]:
```suggestion
   commission_histories = if params[:validator_id]
```
and then you can remove assignments to `commission_histories` inside if-else.

File: app/models/batch.rb [hunk 2]
```diff
@@ -26,6 +26,7 @@
 #
 # Indexes
 #
+#  index_batches_on_network                 (network)
```
Comment (brianlong) [inline app/models/batch.rb:29]: You don't need this index! You already have it with the indexes below.

File: app/models/commission_history.rb [hunk 3]
```diff
@@ -8,13 +8,16 @@
...
 #  epoch_completion  :float(24)
+#  network           :string(191)
 #  created_at        :datetime         not null
 #  updated_at        :datetime         not null
 #  validator_id      :bigint           not null
 #
 # Indexes
 #
-#  index_commission_histories_on_validator_id  (validator_id)
+#  index_commission_histories_on_network                   (network)
```
Comment (brianlong) [inline app/models/commission_history.rb:18]: You don't need this index! You already have it with the indexes below.

File: app/models/commission_history.rb [hunk 4]
```diff
@@ -8,13 +8,16 @@
...
+#  index_commission_histories_on_network_and_validator_id  (network,validator_id)
+#  index_commission_histories_on_validator_id              (validator_id)
```
Comment (brianlong) [inline app/models/commission_history.rb:20]: The three indexes above don't make sense when taken together. You'll never need to run a query on network + validator_id. Normally, you will have an index on [network, account] and then another index on validator_id.

File: app/views/validators/show.html.erb [hunk 5]
```diff
@@ -123,7 +123,10 @@
               @validator.commission.to_i == 100
           %>
           <td class="<%= comm_class %>">
-            <%= @validator.commission rescue 'N/A' %> &percnt;
+            <%= @validator.commission rescue 'N/A' %>&percnt;
+            <%= link_to '(See commission changes)',
+                        commission_histories_path(network: params[:network], validator_id: @validator.id),
+                        class: 'small' %>
```
Comment (brianlong) [inline app/views/validators/show.html.erb:129]: Please make this link visible only when there are commission changes to show. It is bad UX for the user to click a link and then have nothing to show them.

File: app/controllers/public_controller.rb (hunk 1 above)
Comment (kbiala) [reply app/controllers/public_controller.rb:107]: I don't think so.. Still need to assign @validator only if if params[:validator_id] is present

**Review by kbiala — APPROVED** [review]

**Pull request merged**